packages/Logger/src/index.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * MIT License
+
+ * Copyright (c) 2018 Priit Kallas
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+**/
+
+/**
+ * This interface heavily borrows from the `ts-log` package. However, the `trace`
+ * function is not on all of the common loggers that we claim to support so
+ * it's been removed in this copy of this interface.
+ */
+
+/**
+ * Represents a generic logger that could be a simple console, bunyan etc.
+ */
+export interface Logger {
+  debug(message?: any, ...optionalParams: any[]): void;
+  info(message?: any, ...optionalParams: any[]): void;
+  warn(message?: any, ...optionalParams: any[]): void;
+  error(message?: any, ...optionalParams: any[]): void;
+  [x: string]: any;
```

Comment:
What's this for?

Reply:
Similar response to https://github.com/apollographql/apollo-utils/pull/55#discussion_r834781787

It doesn't make sense for us to allow arbitrary properties on the logger internally, but if users want access to those with strong typings they can use their own logger instance which implements `Logger` but might have other facilities.